### DISCUSSION TITLE
RuntimeError: mat1 and mat2 shapes cannot be multiplied

### DISCUSSION THREAD

**Discussion by Kathir0011** [mrdbourke/pytorch-deep-learning#326] [Q&A] [answered]
In pytorch, I'm working on FashionMNIST dataset.
While trying to make prediction with a CNN model, I'm getting the following error
RuntimeError: mat1 and mat2 shapes cannot be multiplied (10x49 and 490x10)

I found that it is occurring because of the final linear layer shape after flattening.
If I change it into this
```
# torch.Size([10, 14, 14])
# torch.Size([10, 7, 7])
# These are output shapes of previous layers

self.output_layer = nn.Sequential(
            nn.Flatten(),
            nn.Linear(in_features=7*7, # before I put in_features=hidden_units*7*7
                      out_features=output_shape)
  )
```
Now, I'm getting an output with a shape of (10, 10). But it has to be (1, 10).
Can someone help?

**Edit**: Solved the issue by specifying Flatten(start_dim=0)

**Comment by toddsp22** [comment]
Looks like it was just a typo
49 doesn't match 490

(10x49 and 490x10)

**Comment by mrdbourke** [comment, marked as answer]
> **Edit:** Solved the issue by specifying Flatten(start_dim=0)

@Kathir0011 glad to hear you got it fixed!

Thank you for updating your answer.
